test/test_unistd.rs

```diff
@@ -400,3 +403,56 @@ fn test_pipe2() {
     let f1 = FdFlag::from_bits_truncate(fcntl(fd1, FcntlArg::F_GETFD).unwrap());
     assert!(f1.contains(FdFlag::FD_CLOEXEC));
 }
+
+// Used in `test_alarm`.
+static mut ALARM_CALLED: bool = false;
+
+// Used in `test_alarm`.
+#[no_mangle]
+#[allow(private_no_mangle_fns)]
+pub extern fn alarm_signal_handler(raw_signal: libc::c_int) {
+    assert_eq!(raw_signal, libc::SIGALRM, "unexpected signal: {}", raw_signal);
+    unsafe { ALARM_CALLED = true };
+}
+
+#[test]
+fn test_alarm() {
+    #[allow(unused_variables)]
+    let m = ::SIGNAL_MTX.lock().expect("Mutex got poisoned by another test");
+
+    // Set an alarm.
+    assert_eq!(alarm(Duration::from_secs(60)), None);
+
+    // Overwriting an alarm should return the old alarm.
+    assert_eq!(alarm(Duration::from_secs(1)), Some(Duration::from_secs(60)));
+    let start = Instant::now();
+
+    let handler = SigHandler::Handler(alarm_signal_handler);
```

Comment:
You should set up the hander _before_ calling `alarm`

Reply:
I know, however this test takes at least a second so I though I add the handler after so it removes some of the overhead, but I'll before it before the call.